apis/keda/v1alpha1/scaledobject_webhook.go

```diff
@@ -318,14 +318,20 @@ func verifyCPUMemoryScalers(incomingSo *ScaledObject, action string, dryRun bool
 				}
 
 				if trigger.Type == cpuString {
-					if !isWorkloadResourceSet(container.Resources, corev1.ResourceCPU) {
+					// Fail if neither pod's container spec has a CPU limit specified, nor a default CPU limit is
+					// specified in LimitRange in the same namespace as the deployment
+					if !isWorkloadResourceSet(container.Resources, corev1.ResourceCPU) &&
+						!isContainerResourceLimitSet(context.Background(), incomingSo.Namespace, corev1.ResourceCPU) {
```

Comment:
we can further simplify this later (I realize it's not in scope for this PR, just a note for myself)

something along the lines of:
```go
if trigger.Type == cpuString || trigger.Type == memoryString {
    // Fail if neither pod's container spec has particular resource limit specified, nor a default limit is
    // specified in LimitRange in the same namespace as the deployment
    resourceType := corev1.ResourceName(trigger.Type)
    if !isWorkloadResourceSet(container.Resources, resourceType) &&
        !isContainerResourceLimitSet(context.Background(), incomingSo.Namespace, resourceType) {
        err := fmt.Errorf("the scaledobject has a %v trigger but the container %s doesn't have the %v request defined", resourceType, container.Name,
        scaledobjectlog.Error(err, "validation error")
        metricscollector.RecordScaledObjectValidatingErrors(incomingSo.Namespace, action, "missing-requests")
        return err
    }
}
```
```

Reply:
Tracking here https://github.com/kedacore/keda/issues/5397